File: Halo/old/(Depreciated)GalaxyGen.py
import numpy as np
import random
import math
import time

t0 = time.time()

txt = np.loadtxt('/home/vamuscari/Coding/Halo/test2.txt')
t1 = time.time()

print("Load Time: " + str((t1 - t0)/60))

Mass = [int(m) for m in txt[:,0]]
MassTF = []
Total = 0


Mmin = 10 * (10 ** 12)
σ = 0.15 

for M in Mass:
    Num = (1/2)* (1 + math.erf((np.log10(M)-np.log10(Mmin))/σ))
    if Num >= random.random():
        MassTF.append([M,1])
        Total += 1
    else:
        MassTF.append([M,0])

print(Total)

File: Halo/old/(Depreciated)GalaxyGen.jl
#For reading Textfile, for error function, for random number.
using NPZ
using SpecialFunctions
using Random

#use package NPZ for reading writing array
#julia uses jld package 

@time begin


txt = npzread("Halo200.npy")

Mass = [m for m in txt[:,1]]
MassTF = [0 0]
Total = 0


Mmin = 10 * (10 ^ 12)
σ = 0.15 



for M in Mass
    Num = (1/2) * (1 + SpecialFunctions.erf((log10(M)-log10(Mmin))/σ))
    if Num >= rand(Float64)
        global MassTF = vcat(MassTF, [M 1])
        global Total += 1
            
    else
        global MassTF = vcat(MassTF, [M 0])
    end
end
println(Total)

end